main.tf

```diff
@@ -102,6 +101,13 @@ module "stacks" {
   before_init    = try(each.value.settings.spacelift.before_init, null) != null ? each.value.settings.spacelift.before_init : var.before_init
   before_perform = try(each.value.settings.spacelift.before_perform, null) != null ? each.value.settings.spacelift.before_perform : var.before_perform
   before_plan    = try(each.value.settings.spacelift.before_plan, null) != null ? each.value.settings.spacelift.before_plan : var.before_plan
+
+  # If `worker_pool_name` is specified for the stack in YAML config AND `var.worker_pool_name_id_map` contains `worker_pool_name` key,
+  # lookup and use the worker pool ID from the map.
+  # Otherwise, use `var.worker_pool_id`.
+  worker_pool_id = try(lookup(var.worker_pool_name_id_map, each.value.settings.spacelift.worker_pool_name), null) != null ? (
+    lookup(var.worker_pool_name_id_map, each.value.settings.spacelift.worker_pool_name)
+  ) : var.worker_pool_id
```

Comment:
This line should be functionally equal to:
```
worker_pool_id = try(var.worker_pool_name_id_map[each.value.settings.spacelift.worker_pool_name], var.worker_pool_id)
```

Reply:
Or do you plan to have explicit `null` value in a map to reset it to default worker pool name somewhere? 

I'd prefer explicitly setting the `worker_pool_name` to default in this case

Reply:
this is good find, thanks @max-lobur 
this line does the same and is shorter
we don't plan to have null values in the map